app/src/main/java/com/example/tasky/core/data/remote/interceptor/JWTInterceptor.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.tasky.core.data.remote.interceptor
+
+import com.example.tasky.authentication.domain.UserCache
+import okhttp3.Interceptor
+import okhttp3.Response
+import javax.inject.Inject
+
+class JWTInterceptor @Inject constructor(
+    private val userCache: UserCache
+) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val token = userCache.getUser()?.token
+        val request = chain.request().newBuilder()
+        token?.let {
+            request.addHeader(authorization, "Bearer $it")
+        }
+        return chain.proceed(request.build())
+    }
+
+    companion object {
+        private const val authorization = "Authorization"
```

Comment:
This will be used only once. So no need for a constant, you could just have it in the code